src/mag_toolkit/calibration/ScienceLayer.py

```diff
@@ -117,34 +154,41 @@ def _from_csv(cls, path: Path):
 
     @classmethod
     def _from_cdf(cls, path: Path):
-        cdf_loaded = pycdf.CDF(str(path))
+        dataset = cdf_to_xarray(str(path), to_datetime=False)
+
+        # cdf_loaded = pycdf.CDF(str(path))
 
-        data = cdf_loaded["vectors"][...]
-        epoch = cdf_loaded["epoch"][...]
+        data = dataset["vectors"].values
+        raw_epoch = dataset["epoch"].values
+        epoch = cdfepoch.to_datetime(raw_epoch)
 
         if data is None or epoch is None:
             raise ValueError("CDF does not contain valid data")
 
         validity = Validity(start=epoch[0], end=epoch[-1])
 
-        sensor = Sensor.MAGO if cdf_loaded.attrs["is_mago"] else Sensor.MAGI
+        sensor = Sensor.MAGO if dataset.attrs["is_mago"][0] == "True" else Sensor.MAGI
 
-        version = int(cdf_loaded.attrs["Data_version"][0][1:])
+        version = int(dataset.attrs["Data_version"][0][1:])
 
         metadata = CalibrationMetadata(
             dependencies=[],
             science=[],
-            creation_timestamp=datetime.now(),
+            creation_timestamp=np.datetime64("now"),
         )
 
         values = [
-            ScienceValue(time=epoch_val, value=datapoint[0:3], range=datapoint[3])
-            for epoch_val, datapoint in zip(epoch, data)
+            ScienceValue(
+                time=epoch_val,
+                value=datapoint[0:3],
+                range=datapoint[3],
+            )
+            for raw_epoch_val, epoch_val, datapoint in zip(raw_epoch, epoch, data)
         ]
 
         return ScienceLayer(
-            id=cdf_loaded.attrs["Logical_file_id"][0],
-            mission=cdf_loaded.attrs["Mission_group"][0],
+            id=dataset.attrs["Logical_file_id"][0],
+            mission=dataset.attrs["Mission_group"][0],
```

Comment:
Any way to reduce the duplication between `_from_cdf` and `_from_csc` - lots of same boilerplate there? Also one creates the class using `ScienceLayer` and the other use `cls` which is just a bit inconsistent?